File: DeanOfficeApp.Api/DeanOfficeApp.Api/Models/Enrollment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DeanOfficeApp.Api.Models
{
    public class Enrollment
    {
        [Key]
        [DatabaseGe‌​nerated(DatabaseGen‌​eratedOption.Identity)]
        public int Id { get; set; }
        public DateTime EnrollmentDate { get; set; }

        [ForeignKey("Student")]
        public int StudentId { get; set; }
        public virtual Student Student { get; set; }

        public int LectureId { get; set; }
        public virtual Lecture Lecture { get; set; }

        public virtual ICollection<Grade> Grades { get; set; }
    }
}

File: deanApp-front/src/app/models/Enrollment.ts
import { Lecture } from './Lecture'
import { Grade } from './Grade'

export class Enrollment {
	id: number
	lectureId: number
	lecture?: Lecture
	studentRecordBookNumber: number
	lectureName: string
	studentName: string
	studentLastName: string
	grades: Grade[]
}
